index.js

```diff
@@ -2255,17 +2255,44 @@ Keys (DM bot)
             }
           } else if (split[0] === prefix + 'login') {
             const id = message.author.id;
-            
-            const code = new Uint32Array(crypto.randomBytes(4).buffer, 0, 1).toString(10).slice(-6);
-            await ddb.putItem({
-              TableName: usersTableName,
-              Item: {
-                email: {S: id + '.code'},
-                code: {S: code},
-              }
-            }).promise();
-            
-            const m = await message.author.send(`Login: https://webaverse.com/login?id=${id}&code=${code}`);
+
+            let {mnemonic} = await _getUser();
+            if (!mnemonic) {
+              const spec = await _genKey();
+              mnemonic = spec.mnemonic;
+            }
+            const wallet = hdkey.fromMasterSeed(bip39.mnemonicToSeedSync(mnemonic)).derivePath(`m/44'/60'/0'/0/0`).getWallet();
+            const address = wallet.getAddressString();
+
+            const currentName = await runSidechainTransaction(mnemonic)('Account', 'getMetadata', address, 'name');
```

Comment:
I don't think this is the right way to query data from the chain. It should be instead:

```
contracts.Account.methods.getMetadata(address, 'avatarId').call();
```

The reason is that `runSidechainTransaction` sends the keys and signals intent to do a write. This call doesn't write though, and has nothing to do with the account; it's a function call read.